### DISCUSSION TITLE
How to use IntelliJ Idea and/or VSCode IDE for mill and sbt builds at the same time in the same directory/project?

### DISCUSSION THREAD

**Discussion by jk-1** [com-lihaoyi/mill#1530] [Q&A]
I try to transform some of my sbt builds to mill. It would help the transition if I could use IDE for both mill and sbt project at the same time in the same directory during the transition time.

Is it possible for IntelliJ Idea or VSCode + metals to open and use two different projects (build.sbt, build.sc) in the same directory and if yes then how to do it?

One way to achieve this (which I have not yet tried) might be to use both VSCode and IntelliJ Idea at the same time, one for sbt build and other for mill build. Any ideas which one to use for mill build and which one to use for sbt build? Why?

**Comment by lefou** [comment]
I tried this myself while experimenting with mill BSP support. I edited mill via `mill.scalalib.GenIdea/idea` generated project in IntelliJ IDEA and opened the same project via VSCode + metals (via Bloop and/or BSP). I have not much experience with sbt in either IDE (beside of remembering that I had to wait way too long before it was usable), so you need to experiment yourself which combination works best for your use case. Speaking about mill, I think the IntelliJ generator is the best option if you also want to edit the `build.sc` files. But you need to re-generate it whenever you make significant changes to your setup.
